OpenRA.Mods.Common/Lint/CheckNotifications.cs

```diff
@@ -0,0 +1,68 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Lint
+{
+	class CheckNotifications : ILintRulesPass
+	{
+		public void Run(Action<string> emitError, Action<string> emitWarning, Ruleset rules)
+		{
+			foreach (var actorInfo in rules.Actors)
+			{
+				foreach (var traitInfo in actorInfo.Value.TraitInfos<ITraitInfo>())
+				{
+					var fields = traitInfo.GetType().GetFields().Where(f => f.HasAttribute<NotificationReferenceAttribute>());
+					foreach (var field in fields)
+					{
+						var notifications = LintExts.GetFieldValues(traitInfo, field, emitError);
+						foreach (var notification in notifications)
+						{
+							if (string.IsNullOrEmpty(notification))
+								continue;
+
+							CheckVoices(actorInfo.Value, emitError, rules, notification);
+						}
+					}
+				}
+			}
+		}
+
+		void CheckVoices(ActorInfo actorInfo, Action<string> emitError, Ruleset rules, string notification)
```

Comment:
Again, please rename this to CheckNotifications. It's not voices per se what you're checking for.